lib/client.js

```diff
@@ -23,6 +41,15 @@ var _defaultOptions = {
     certificate:  process.env.TASKCLUSTER_CERTIFICATE
   },
 
+  timeout: 30 * 1000, // 30 seconds
+
+  // Http agent options (https://nodejs.org/api/http.html#http_new_agent_options)
+  agent: {
```

Comment:
Are you sure this is a good idea... Basically you'll never use `DEFAULT_AGENTS` ... Because of:
https://github.com/taskcluster/taskcluster-client/pull/17/files#diff-50cfa59973c04321b5da0c6da0fdf4feL91